*** amsclass.dtx.org	2009-07-02 18:05:00.000000000 +0200
--- amsclass.dtx	2009-09-05 03:09:04.000000000 +0200
***************
*** 36,42 ****
  \author{American Mathematical Society\\Michael Downes\\
    updated by Barbara Beeton}
  \date{Version \fileversion, \filedate}
! \hDocInput{amsclass.dtx}
  \end{document}
  %</driver>
  % \fi
--- 36,42 ----
  \author{American Mathematical Society\\Michael Downes\\
    updated by Barbara Beeton}
  \date{Version \fileversion, \filedate}
! \DocInput{amsclass.dtx}
  \end{document}
  %</driver>
  % \fi
