*** amsfndoc.tex.org	2009-02-22 20:46:31.000000000 +0100
--- amsfndoc.tex	2013-02-14 14:58:48.915336160 +0100
***************
*** 105,111 ****
    instead of sizes 5,6,7,8,9,10 when the Y\&Y/Blue Sky PostScript outline
    fonts are used
  \itemitem{--} \filename{*.fd}, font definition files
! \item{} For details, see Section 3 on using AMSFonts with \LaTeX{}.
  
  \item{\bull} Macro files for using these fonts with plain \TeX{}:
  \itemitem{--} \filename{amssym.tex}, a file defining names for the symbols
--- 105,111 ----
    instead of sizes 5,6,7,8,9,10 when the Y\&Y/Blue Sky PostScript outline
    fonts are used
  \itemitem{--} \filename{*.fd}, font definition files
! \item{} For details, see \MakeLink{section.3}{Section 3} on using AMSFonts with \LaTeX{}.
  
  \item{\bull} Macro files for using these fonts with plain \TeX{}:
  \itemitem{--} \filename{amssym.tex}, a file defining names for the symbols
***************
*** 166,172 ****
  \parskip=0pt
  \item{\bull} ``{\tt cm}'': These fonts are based directly on the
    specifications for Knuth's Computer Modern fonts, as described
!   in Volume~E of \CandT{} [DEK86E].
  \item{\bull} ``{\tt eu}'': These are members of the Euler family,
    described below.
  \item{\bull} ``{\tt ms}'': These fonts of math symbols were developed
--- 166,172 ----
  \parskip=0pt
  \item{\bull} ``{\tt cm}'': These fonts are based directly on the
    specifications for Knuth's Computer Modern fonts, as described
!   in Volume~E of \CandT{} \Cite[DEK86E].
  \item{\bull} ``{\tt eu}'': These are members of the Euler family,
    described below.
  \item{\bull} ``{\tt ms}'': These fonts of math symbols were developed
***************
*** 211,224 ****
  eighteenth century.  Zapf's designs were rendered in \MF{} code by graduate
  students at Stanford, working under Knuth's direction; the process by which
  the \MF{} fonts were implemented is described in a report by David Siegel
! [DRS85].  The Euler fonts were designed to be used as math symbols; they are
  not intended for setting running text.
  
  The Fraktur face of the Euler family has been used in production by the
  AMS since it became available.  However, no extensive test or use had
  been made of the script or cursive until Knuth decided that they should be
  used in a textbook, {\sl Concrete Mathematics}, written by him and two
! co-authors [GKP88].  During the course of preparing that book, a number of
  errors, particularly in spacing parameters affecting the placement of sub-
  and superscripts, were discovered.  All these errors have been corrected in
  the medium-weight versions of the Euler fonts (almost no boldface symbols
--- 211,224 ----
  eighteenth century.  Zapf's designs were rendered in \MF{} code by graduate
  students at Stanford, working under Knuth's direction; the process by which
  the \MF{} fonts were implemented is described in a report by David Siegel
! \Cite[DRS85].  The Euler fonts were designed to be used as math symbols; they are
  not intended for setting running text.
  
  The Fraktur face of the Euler family has been used in production by the
  AMS since it became available.  However, no extensive test or use had
  been made of the script or cursive until Knuth decided that they should be
  used in a textbook, {\sl Concrete Mathematics}, written by him and two
! co-authors \Cite[GKP88].  During the course of preparing that book, a number of
  errors, particularly in spacing parameters affecting the placement of sub-
  and superscripts, were discovered.  All these errors have been corrected in
  the medium-weight versions of the Euler fonts (almost no boldface symbols
***************
*** 227,233 ****
  the integral sign, was too slanted to be attractive with Euler, and
  consequently he prepared a new (partial) extension font for use with Euler.
  Knuth described his experience with the Euler fonts in a \TUB\/ article
! [DEK89].  In the article he also identified the macros he used and where they
  can be obtained.
  
  The Euler fonts are sparsely populated; only the alphabetic locations are
--- 227,233 ----
  the integral sign, was too slanted to be attractive with Euler, and
  consequently he prepared a new (partial) extension font for use with Euler.
  Knuth described his experience with the Euler fonts in a \TUB\/ article
! \Cite[DEK89].  In the article he also identified the macros he used and where they
  can be obtained.
  
  The Euler fonts are sparsely populated; only the alphabetic locations are
***************
*** 243,249 ****
  includes Greek), symbol, and math extension fonts are included in standard
  distributions of \TeX{}.  Since these symbols are often needed in
  mathematics, other sizes have been constructed, using the principles
! demonstrated in Knuth's {\sl Computer Modern Typefaces\/} [DEK86E], and
  included in the AMSFonts collection.
  
  
--- 243,249 ----
  includes Greek), symbol, and math extension fonts are included in standard
  distributions of \TeX{}.  Since these symbols are often needed in
  mathematics, other sizes have been constructed, using the principles
! demonstrated in Knuth's {\sl Computer Modern Typefaces\/} \Cite[DEK86E], and
  included in the AMSFonts collection.
  
  
***************
*** 318,324 ****
  
  The dummy font is used in \AmSTeX{} to implement ``syntax checking''.
  (Syntax checking is activated by the |\printoptions| command as
! described in \JoT{} [MDS86].) In this mode, the dummy font replaces all
  the usual ``printing'' fonts, so that \TeX{} never accumulates any text
  to be set, and never tries to write out a page, but in the process of
  reading the input file, checks all control sequences for syntactic
--- 318,324 ----
  
  The dummy font is used in \AmSTeX{} to implement ``syntax checking''.
  (Syntax checking is activated by the |\printoptions| command as
! described in \JoT{} \Cite[MDS86].) In this mode, the dummy font replaces all
  the usual ``printing'' fonts, so that \TeX{} never accumulates any text
  to be set, and never tries to write out a page, but in the process of
  reading the input file, checks all control sequences for syntactic
***************
*** 328,334 ****
  overfull boxes cannot be detected until setting actually occurs.
  
  
! \subsection{2.6} PostScript (Type 1) implementation of AMSFonts
  
  The AMSFonts are available in Type~1 outline form as an alternative
  to the bitmap images generated by \MF.  The Type~1 files are not
--- 328,334 ----
  overfull boxes cannot be detected until setting actually occurs.
  
  
! \subsection{2.7} PostScript (Type 1) implementation of AMSFonts
  
  The AMSFonts are available in Type~1 outline form as an alternative
  to the bitmap images generated by \MF.  The Type~1 files are not
***************
*** 480,486 ****
  
  The \pkg{amssymb} package defines math symbol commands for all the extra
  math symbols in the \fn{msam} and \fn{msbm} fonts, as listed in the
! table in Section 7.  Thus if you want to use the \cn{blacktriangle}
  $\blacktriangle$ and \cn{nsubseteq} $\nsubseteq$ symbols, the easiest
  way is to put the statement
  \begintt
--- 480,486 ----
  
  The \pkg{amssymb} package defines math symbol commands for all the extra
  math symbols in the \fn{msam} and \fn{msbm} fonts, as listed in the
! table in \MakeLink{section.7}{Section 7}.  Thus if you want to use the \cn{blacktriangle}
  $\blacktriangle$ and \cn{nsubseteq} $\nsubseteq$ symbols, the easiest
  way is to put the statement
  \begintt
***************
*** 491,497 ****
  If you want more selective access to the math symbols in those fonts
  you can use the \pkg{amsfonts} package instead and define math symbol
  commands individually using \LaTeX{}'s \cn{DeclareMathSymbol}
! (cf.\ [LFG] or [GMS94]):
  \begintt
  \usepackage{amsfonts}
  \DeclareMathSymbol{\blacktriangle}{\mathord}{AMSa}{"4E}
--- 491,497 ----
  If you want more selective access to the math symbols in those fonts
  you can use the \pkg{amsfonts} package instead and define math symbol
  commands individually using \LaTeX{}'s \cn{DeclareMathSymbol}
! (cf.\ \Cite[LFG] or \Cite[GMS94]):
  \begintt
  \usepackage{amsfonts}
  \DeclareMathSymbol{\blacktriangle}{\mathord}{AMSa}{"4E}
***************
*** 511,517 ****
  Since \cn{DeclareMathSymbol} is used in the \pkg{amssymb} package, the
  definitions for particular symbols can be borrowed from there (file
  \fn{amssymb.sty}).  Alternatively, the values can be obtained from the
! tables in Section 7.2, as follows:
  
  \begingroup
  \raggedright
--- 511,517 ----
  Since \cn{DeclareMathSymbol} is used in the \pkg{amssymb} package, the
  definitions for particular symbols can be borrowed from there (file
  \fn{amssymb.sty}).  Alternatively, the values can be obtained from the
! tables in \MakeLink{subsection.7.2}{Section 7.2}, as follows:
  
  \begingroup
  \raggedright
***************
*** 568,574 ****
  rely on general mechanisms provided by \LaTeX{}.
  
  If you require these fonts, you can consult the \LaTeX{} documentation
! [LFG], [L94] and use directly the commands described there, such as
  \cn{DeclareFontFamily}, \cn{DeclareFontShape}, and \cn{symbol}.  A scheme
  for accessing letters that don't correspond to the 26-letter Latin alphabet
  will be needed; the file \fn{cyracc.def} can be used as a starting point.
--- 568,574 ----
  rely on general mechanisms provided by \LaTeX{}.
  
  If you require these fonts, you can consult the \LaTeX{} documentation
! \Cite[LFG], \Cite[L94] and use directly the commands described there, such as
  \cn{DeclareFontFamily}, \cn{DeclareFontShape}, and \cn{symbol}.  A scheme
  for accessing letters that don't correspond to the 26-letter Latin alphabet
  will be needed; the file \fn{cyracc.def} can be used as a starting point.
***************
*** 586,593 ****
  nonalphabetic symbols.
  
  Instructions for using the fonts of the AMSFonts collection with \AmSTeX{}
! are also given in the {\sl User's Guide to \AmSTeX{} Version~2.2\/} [AMS01]
! and in Appendix~G of editions of \JoT{} [MDS90] dated 1990 or later.
  
  Additional fonts to be used with \AmSTeX{} should be specified at the top
  of the document input file, in what is known as the ``preamble''.  The
--- 586,593 ----
  nonalphabetic symbols.
  
  Instructions for using the fonts of the AMSFonts collection with \AmSTeX{}
! are also given in the {\sl User's Guide to \AmSTeX{} Version~2.2\/} \Cite[AMS01]
! and in Appendix~G of editions of \JoT{} \Cite[MDS90] dated 1990 or later.
  
  Additional fonts to be used with \AmSTeX{} should be specified at the top
  of the document input file, in what is known as the ``preamble''.  The
***************
*** 612,619 ****
  commands |\loadmsam|, |\loadmsbm|, and |\loadeufm| will load the
  medium-weight versions of the two extra symbol fonts and Euler Fraktur
  respectively.  The command |\newsymbol| can then be used to define just
! those symbols that are needed; its use is described in Section 7,
! {\bf Using the extra symbols}.%
  \footnote{*}{Additional fonts from the AMSFonts collection can be accessed
    easily in \AmSTeX{}.
    However, users should be aware that \TeX{} limits the number of
--- 612,619 ----
  commands |\loadmsam|, |\loadmsbm|, and |\loadeufm| will load the
  medium-weight versions of the two extra symbol fonts and Euler Fraktur
  respectively.  The command |\newsymbol| can then be used to define just
! those symbols that are needed; its use is described in \MakeLink{section.7}{Section 7,
! {\bf Using the extra symbols}}.%
  \footnote{*}{Additional fonts from the AMSFonts collection can be accessed
    easily in \AmSTeX{}.
    However, users should be aware that \TeX{} limits the number of
***************
*** 698,705 ****
  \endgroup
  \noindent
  These facilities are described in more detail in the {\sl User's Guide to
! \AmSTeX{} Version~2.2\/} [AMS01] and editions of \Joy\/ published in 1990
! or later [MDS90].
  
  
  \subsection{4.3} Computer Modern math extension font
--- 698,705 ----
  \endgroup
  \noindent
  These facilities are described in more detail in the {\sl User's Guide to
! \AmSTeX{} Version~2.2\/} \Cite[AMS01] and editions of \Joy\/ published in 1990
! or later \Cite[MDS90].
  
  
  \subsection{4.3} Computer Modern math extension font
***************
*** 708,714 ****
  smaller than ten-point and in sub- and superscripts.  They are provided
  automatically for these environments in the preprint style.  If you are
  not using the preprint style, you can use the font definitions in either
! \filename{amsppt.sty} or Appendix~E of \TB\/ [DEK86A] as a model.
  
  
  \subsection{4.4} Extra symbols
--- 708,714 ----
  smaller than ten-point and in sub- and superscripts.  They are provided
  automatically for these environments in the preprint style.  If you are
  not using the preprint style, you can use the font definitions in either
! \filename{amsppt.sty} or Appendix~E of \TB\/ \Cite[DEK86A] as a model.
  
  
  \subsection{4.4} Extra symbols
***************
*** 720,726 ****
  |\loadmsbm| in the preamble of your document.  If you load the fonts
  separately, a few symbols will be defined when one of the fonts is loaded,
  but most must be defined using the |\newsymbol| command before they can be
! used.  See Section 7, {\bf Using the extra symbols}, for
  information on both the symbol names and on using |\newsymbol| to define them.
  
  
--- 720,726 ----
  |\loadmsbm| in the preamble of your document.  If you load the fonts
  separately, a few symbols will be defined when one of the fonts is loaded,
  but most must be defined using the |\newsymbol| command before they can be
! used.  See \MakeLink{section.7}{Section 7, {\bf Using the extra symbols}}, for
  information on both the symbol names and on using |\newsymbol| to define them.
  
  
***************
*** 729,735 ****
  Cyrillic is not referred to in the \AmSTeX{} files as distributed.
  The cyrillic fonts included in AMSFonts are intended for use mainly in text,
  not as symbols in math.  Detailed instructions for loading and using cyrillic
! appear below in Section 6, {\bf Using cyrillic}.
  
  
  \subsection{4.6} Caps/small caps
--- 729,735 ----
  Cyrillic is not referred to in the \AmSTeX{} files as distributed.
  The cyrillic fonts included in AMSFonts are intended for use mainly in text,
  not as symbols in math.  Detailed instructions for loading and using cyrillic
! appear below in \MakeLink{section.6}{Section 6, {\bf Using cyrillic}}.
  
  
  \subsection{4.6} Caps/small caps
***************
*** 737,743 ****
  Caps/small caps are loaded automatically by the \AmSTeX{} preprint style
  for use in ten-point and eight-point text.  If you are not using the preprint
  style, you can use the font definitions in either \filename{amsppt.sty}
! or Appendix~E of \TB\/ [DEK86A] as a model.
  
  
  \subsection{4.7} Dummy font
--- 737,743 ----
  Caps/small caps are loaded automatically by the \AmSTeX{} preprint style
  for use in ten-point and eight-point text.  If you are not using the preprint
  style, you can use the font definitions in either \filename{amsppt.sty}
! or Appendix~E of \TB\/ \Cite[DEK86A] as a model.
  
  
  \subsection{4.7} Dummy font
***************
*** 853,864 ****
  \subsection{5.4} Extra symbols
  
  Detailed instructions for accessing the \fontname{msam} and \fontname{msbm}
! fonts are given in Section 7, {\bf Using the extra symbols}.
  
  
  \subsection{5.5} Cyrillic
  
! See Section 6, {\bf Using cyrillic}, for instructions.
  
  
  \subsection{5.6} Caps/small caps
--- 853,864 ----
  \subsection{5.4} Extra symbols
  
  Detailed instructions for accessing the \fontname{msam} and \fontname{msbm}
! fonts are given in \MakeLink{section.7}{Section 7, {\bf Using the extra symbols}}.
  
  
  \subsection{5.5} Cyrillic
  
! See \MakeLink{section.6}{Section 6, {\bf Using cyrillic}}, for instructions.
  
  
  \subsection{5.6} Caps/small caps
***************
*** 933,939 ****
  \endtt
  If you require cyrillic text in more than one size, you must take a
  different approach in defining |\cyr|.  An appropriate model appears in
! Appendix~E of \TB\/ [DEK86A], pages 414--15.  The definition of |\cyr|
  should be incorporated into size-specific macros such as |\tenpoint| and
  |\eightpoint| similarly to what is done there for |\bf|.  Don't forget to
  include the command |\cyracc| in the definition.
--- 933,939 ----
  \endtt
  If you require cyrillic text in more than one size, you must take a
  different approach in defining |\cyr|.  An appropriate model appears in
! Appendix~E of \TB\/ \Cite[DEK86A], pages 414--15.  The definition of |\cyr|
  should be incorporated into size-specific macros such as |\tenpoint| and
  |\eightpoint| similarly to what is done there for |\bf|.  Don't forget to
  include the command |\cyracc| in the definition.
***************
*** 1046,1052 ****
    used extensively by the AMS cyrillic input conventions, are not easily
    specified to \TeX's hyphenation mechanism.  Another approach to
    hyphenation, requiring some changes to the cyrillic \filename{.tfm}
!   files, has been described by Dimitri Vulis in a \TUB\/ article [DLV89].
  
  
  \subsection{6.3} Cyrillic in math
--- 1046,1052 ----
    used extensively by the AMS cyrillic input conventions, are not easily
    specified to \TeX's hyphenation mechanism.  Another approach to
    hyphenation, requiring some changes to the cyrillic \filename{.tfm}
!   files, has been described by Dimitri Vulis in a \TUB\/ article \Cite[DLV89].
  
  
  \subsection{6.3} Cyrillic in math
***************
*** 1073,1079 ****
  font family specifications in |\addto\tenpoint|, as shown above.
  
  If other base text sizes are used besides ten point, the suggestions given
! above in Section 6.1, {\bf Making cyrillic available}, apply here as well.
  
  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  
--- 1073,1079 ----
  font family specifications in |\addto\tenpoint|, as shown above.
  
  If other base text sizes are used besides ten point, the suggestions given
! above in \MakeLink{subsection.6.1}{Section 6.1, {\bf Making cyrillic available}}, apply here as well.
  
  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
  
***************
*** 1126,1133 ****
  accessible to their \TeX{} jobs with the least possible fuss.  For \AmSTeX{}
  users, these fonts are available automatically with the preprint style, and
  other methods of loading them for use with \AmSTeX{} are described above.
! To load these fonts with \LaTeX{} or \AmS-\LaTeX{}, see Section 3.4,
! {\bf Extra math symbols}.
  
  If you are not using \AmSTeX{} or \LaTeX{}, the easiest method of loading
  these fonts and defining the control sequences for accessing the symbols is
--- 1126,1133 ----
  accessible to their \TeX{} jobs with the least possible fuss.  For \AmSTeX{}
  users, these fonts are available automatically with the preprint style, and
  other methods of loading them for use with \AmSTeX{} are described above.
! To load these fonts with \LaTeX{} or \AmS-\LaTeX{}, see \MakeLink{subsection.3.4}{Section 3.4,
! {\bf Extra math symbols}}.
  
  If you are not using \AmSTeX{} or \LaTeX{}, the easiest method of loading
  these fonts and defining the control sequences for accessing the symbols is
***************
*** 1413,1419 ****
  \bibitem [DLV89] Dimitri Vulis, ``Notes on Russian \TeX'',
    {\sl \TUB\/} {\bf10} (1989), no.~3, 332--36.
  
! \bibitem [DRS86] David R Siegel, {\sl The Euler Project at Stanford},
    Computer Science Department, Stanford University, 1985.
  
  \bibitem [GKP88] Ronald L. Graham, Donald E. Knuth, and Oren Patashnik,
--- 1413,1419 ----
  \bibitem [DLV89] Dimitri Vulis, ``Notes on Russian \TeX'',
    {\sl \TUB\/} {\bf10} (1989), no.~3, 332--36.
  
! \bibitem [DRS85] David R Siegel, {\sl The Euler Project at Stanford},
    Computer Science Department, Stanford University, 1985.
  
  \bibitem [GKP88] Ronald L. Graham, Donald E. Knuth, and Oren Patashnik,
