*** encguide.tex.org	2009-09-24 17:00:50.000000000 +0200
--- encguide.tex	2011-07-01 22:00:17.000000000 +0200
***************
*** 1226,1232 ****
  \newblock {\em {TUG}boat}, 11(4):514--516, 1990.

  \bibitem{fontinst} \emph{The \Pkg{fontinst} package}:
!   \textlangle CTAN\textrangle\url{/fonts/utilities/fontinst}.

  \bibitem{Rei:TB17-2-102} Fukui Rei:
    \emph{\textsl{TIPA}: A system for processing phonetic
--- 1226,1234 ----
  \newblock {\em {TUG}boat}, 11(4):514--516, 1990.

  \bibitem{fontinst} \emph{The \Pkg{fontinst} package}:
!   \href{ftp://ftp.ctan.org/tex-archive/fonts/utilities/fontinst}{%
!     \textlangle CTAN\textrangle\nolinkurl{/fonts/utilities/fontinst}%
!   }.

  \bibitem{Rei:TB17-2-102} Fukui Rei:
    \emph{\textsl{TIPA}: A system for processing phonetic
***************
*** 1266,1272 ****
  \newblock Addison-Wesley, Reading, Massachusetts, second edition, 1994.

  \bibitem{LH-Fonts} \emph{The \Pkg{lh}-Fonts for Cyrillic}:
!   \textlangle CTAN\textrangle\url{/fonts/cyrillic/lh}.

  \bibitem{A-W:MG2004}
  Frank Mittelbach and Michel Goossens.
--- 1268,1276 ----
  \newblock Addison-Wesley, Reading, Massachusetts, second edition, 1994.

  \bibitem{LH-Fonts} \emph{The \Pkg{lh}-Fonts for Cyrillic}:
!   \href{ftp://ftp.ctan.org/tex-archive/fonts/cyrillic/lh}{%
!     \textlangle CTAN\textrangle\nolinkurl{/fonts/cyrillic/lh}%
!   }.

  \bibitem{A-W:MG2004}
  Frank Mittelbach and Michel Goossens.
