*** logmac.tex.org	1991-10-19 05:33:44.000000000 +0100
--- logmac.tex	2008-07-11 01:00:43.000000000 +0200
***************
*** 42,54 ****
  \catcode`\	=12 % tabs are ordinary characters
  \catcode`@=\active  % @ signs denote a TeX section reference
  {\obeylines \gdef@#1
!  {\global\setbox\texsec=\hbox{\S$#1$}\lookahead}}
  \chardef\@=`\@
  {\catcode`X=\active \gdef X{\X}}
  \def\X{{\eightbf X}}
  \def\lightX{\hbox{\eightrm X}}
! \def\>#1\relax{\global\setbox\predec=\hbox{\let\X=\lightX
!   $#1\mapsto{}$}} % previous bug ref
  
  \def\checkvalid#1{\expandafter\ifx\csname#1!\endcsname\okay\else\invalidcode\fi}
  \def\okay{\okay}
--- 42,103 ----
  \catcode`\	=12 % tabs are ordinary characters
  \catcode`@=\active  % @ signs denote a TeX section reference
  {\obeylines \gdef@#1
!   {%
!     \global\setbox\texsec=\hbox{%
!       \S$\texlinks{#1}$%
!     }%
!     \lookahead %
!   }%
! }
! \def\texlinks#1{%
!   \dotexlinks#1,\END
! }
! \def\dotexlinks#1,#2\END{%
!   {%
!     \pdfstartlink attr{%
!       /Border[0 0 0]%
!     } user {%
!       /Subtype/Link%
!       /F 4%
!       /A<<%
!         /S/GoToR%
!         /F(../tex/tex.pdf)%
!         /D(#1)%
!       >>%
!     }\relax
!     \TeXLinkColor #1\Black
!     \pdfendlink
!   }%
!   \ifx\eject#2\eject
!   \else
!     ,%
!     \dotexlinks#2\END
!   \fi
! }
! \def\TeXLinkColor{%
!   \pdfliteral{0 0 1 rg}%
! }
  \chardef\@=`\@
  {\catcode`X=\active \gdef X{\X}}
  \def\X{{\eightbf X}}
  \def\lightX{\hbox{\eightrm X}}
! \def\>#1\relax{%
!   \global\setbox\predec=\hbox{%
!     \let\X=\lightX
!     $%
!     \begingroup
!       \def\hbox##1{##1}%
!       \def\eightrm{}%
!       \xdef\linkname{#1}%
!     \endgroup
!     \pdfstartlink attr{/Border[0 0 0]} goto name{\linkname}\relax
!     \LinkColor#1\Black
!     \pdfendlink
!     \mapsto{}$%
!   }%
! } % previous bug ref
! \def\LinkColor{\pdfliteral{1 0 0 rg}}
! \def\Black{\pdfliteral{0 g}}
  
  \def\checkvalid#1{\expandafter\ifx\csname#1!\endcsname\okay\else\invalidcode\fi}
  \def\okay{\okay}
***************
*** 82,90 ****
    \global\setbox\code=\hbox to\codesize{\enspace
     \hskip0pt plus 3fil\rm#1\hskip0pt plus1fil}%
    \begingroup\rm\catcode`X=\active\itemnumber}
! \def\itemnumber#1.{\catcode`X=11\setbox0=\hbox to\itemnosize{\bf\hss#1\relax}%
!   \setbox2=\hbox{\bf#1\relax}\ifdim\wd2>\wd0 \setbox0=\box2 \fi
!   \textindent{\hfil\box\predec\box0}}
  \def\commentline{\fin\afterassignment\begincomment\let\next}
  \def\begincomment{\begingroup\let\tt=\sltt
    \let\MF=\slMF\sl\textindent{\hfil\copy\cbox}}
--- 131,172 ----
    \global\setbox\code=\hbox to\codesize{\enspace
     \hskip0pt plus 3fil\rm#1\hskip0pt plus1fil}%
    \begingroup\rm\catcode`X=\active\itemnumber}
! \def\itemnumber#1.{%
!   \catcode`X=11 %
!   \setbox0=\hbox to\itemnosize{%
!     \bf\hss#1\relax
!   }%
!   \setbox2=\hbox{\bf#1\relax}%
!   \ifdim\wd2>\wd0 %
!     \setbox0=\box2 %
!   \fi
!   \textindent{%
!     \hfil
!     \box\predec
!     \box0 %
!     \rlap{%
!       \enspace
!       \kern-\parindent
!       \vbox to 0pt{%
!         \kern-2.5ex\relax
!         \def\getnum##1\>##2\END{##1}%
!         \def\eightbf{}%
!         \edef\linkname{\getnum#1\>\END}%
!         \def\firstofone##1{##1}%
!         \edef\linkname{\expandafter\firstofone\linkname}%
!         \expandafter\ifx\csname :\linkname:\endcsname\empty
!           \immediate\write16{%
!             * Warning: Destination `\linkname' already used!%
!           }%
!         \else
!           \expandafter\xdef\csname :\linkname:\endcsname{}%
!           \pdfdest name{\linkname}fitbh\relax
!           \kern2.5ex\relax
!         \fi
!       }%
!     }%
!   }%
! }
  \def\commentline{\fin\afterassignment\begincomment\let\next}
  \def\begincomment{\begingroup\let\tt=\sltt
    \let\MF=\slMF\sl\textindent{\hfil\copy\cbox}}
