*** source2e.tex.org	2009-09-25 09:56:20.000000000 +0200
--- source2e.tex	2010-05-04 22:17:43.000000000 +0200
***************
*** 71,76 ****
--- 71,81 ----

  \documentclass{ltxdoc}

+ \newdimen\OldHeadHeight
+ \OldHeadHeight=\headheight
+ \newdimen\OldHeadSep
+ \OldHeadSep=\headsep
+
  \listfiles

  % Do not index some TeX primitives, and some common plain TeX commands.
***************
*** 301,306 ****
--- 306,325 ----
  \clearpage
  \pagestyle{headings}

+ \makeatletter
+ \headheight=\OldHeadHeight
+ \headsep=\OldHeadSep
+ \addtolength{\textheight}{-\headheight}
+ \addtolength{\textheight}{-\headsep}
+ \addtolength{\textheight}{\footskip}
+ \newdimen\OldFootSkip
+ \OldFootSkip=\footskip
+ \footskip=\z@
+ \vsize=\textheight
+ \@colht\textheight
+ \@colroom\textheight
+ \makeatother
+
  % Make TeX shut up.
  \hbadness=10000
  \newcount\hbadness
***************
*** 315,320 ****
--- 334,359 ----

  \clearpage

+ \makeatletter
+ \footskip=85pt
+ \addtolength{\footskip}{\OldFootSkip}
+ \addtolength{\footskip}{-\baselineskip}
+ \addtolength{\textheight}{-\footskip}
+ \vsize=\textheight
+ \@colht=\textheight
+ \@colroom\textheight
+ \newsavebox\FileKeyBox
+ \g@addto@macro\index@prologue{%
+   \sbox\FileKeyBox{%
+     \parbox[b]{\textwidth}{\strut\footnotesize
+     \raggedright{\bfseries File Key:} \filekey}%
+   }%
+   \typeout{FileKeyBox: ht=\the\ht\FileKeyBox, dp=\the\dp\FileKeyBox}%
+   \gdef\@oddfoot{\copy\FileKeyBox}%
+   \let\@evenfoot\@oddfoot
+ }%
+ \makeatother
+
  % makeindex needs a symbol between the parts of composite page numbers
  % but we dont want one, so:
  \typeout{%
