*** tlc2.err.org	2009-09-24 17:00:50.000000000 +0200
--- tlc2.err	2011-07-01 20:28:10.000000000 +0200
***************
*** 302,308 ****
  
  \documentclass{ttcterrata}[2005/11/10]  % we want new class
  
! \usepackage[T1]{fontenc}
  \let\u\underline          % shortcut for this file
  \newcommand\meta[1]{$\langle$\textit{#1\/}$\rangle$}
  
--- 302,308 ----
  
  \documentclass{ttcterrata}[2005/11/10]  % we want new class
  
! \usepackage{fontspec} % \usepackage[T1]{fontenc}
  \let\u\underline          % shortcut for this file
  \newcommand\meta[1]{$\langle$\textit{#1\/}$\rangle$}
  
