*** tripman.tex.org	1992-01-14 00:55:40.000000000 +0100
--- tripman.tex	2014-02-03 20:51:41.181649319 +0100
***************
*** 22,27 ****
--- 22,42 ----
  % if you're unlucky it will appear on a page by itself!
  {\obeyspaces\global\let =\ }
  
+ \pdfinfo{%
+   /Title(A torture test for TeX)%
+   /Author(Donald E. Knuth)%
+ }
+ \pdfcatalog{%
+   /PageLabels<<%
+     /Nums[0<</P(0)>>1<</S/D/St 1>>]%
+   >>%
+   /PageMode/UseOutlines
+ }
+ \pdfpagewidth=7.5in\relax % 8.5in - 2*.5in
+ \pdfhorigin=.5in\relax
+ \pdfpageheight=10in\relax % 11in - 2*.5in
+ \pdfvorigin=.5in\relax
+ 
  \output{\shipout\box255\global\advance\pageno by 1} % for the title page only
  \null
  \vfill
***************
*** 57,64 ****
      \box255}
    \global\advance\pageno by 1}
  \let\runninghead=\mark
! \outer\def\section#1.{\noindent{\bf#1.}\quad
!   \runninghead{\uppercase{#1} }\ignorespaces}
  
  \section Introduction.
  People often think that their programs are ``debugged'' when large applications
--- 72,117 ----
      \box255}
    \global\advance\pageno by 1}
  \let\runninghead=\mark
! 
! \newcount\destcount
! \destcount=1\relax
! 
! \outer\def\section#1.{%
!   \noindent
!   \vbox to 0pt{%
!     \kern-2.5ex\relax
!     \pdfdest num\destcount fitbh\relax
!     \def\.##1{##1}%
!     \def\TeX{TeX}%
!     \pdfoutline goto num\destcount{#1}\relax
!     \global\advance\destcount by 1\relax
!     \kern2.5ex\relax
!   }%
!   {\bf#1.}\quad
!   \runninghead{\uppercase{#1} }\ignorespaces
! }
! 
! \def\LinkColor{%
!   \pdfliteral{1 0 0 rg}%
! }
! \def\Black{%
!   \pdfliteral{0 g}%
! }
! \def\appA{2}
! \def\appB{3}
! \def\appC{4}
! \def\appD{5}
! \def\appE{6}
! \def\appF{7}
! \def\appG{8}
! \def\appH{9}
! 
! \def\appref#1{%
!   \leavevmode
!   \pdfstartlink attr{/Border[0 0 0]} goto num\csname app#1\endcsname\relax
!   \LinkColor\hbox{Appendix}~#1\Black
!   \pdfendlink
! }
  
  \section Introduction.
  People often think that their programs are ``debugged'' when large applications
***************
*** 170,208 ****
  The remainder of this report consists of appendices, whose contents ought
  to be described briefly here:
  
! Appendix A explains in detail how to carry out a test of \TeX, given
  a tape that contains copies of the other appendices.
  
! Appendix B is \.{TRIP.TEX}, the fiendish test file that has already
  been mentioned. People who think that they understand \TeX\ are challenged
  to see if they know what \TeX\ is supposed to do with this file.
  People who know only a little about \TeX\ might still find it
  interesting to study Appendix~B, just to get some insights into the
  methodology advocated here.
  
! Appendix C is \.{TRIP.PL}, the property-list description of a
  special font called \.{trip}. This is the only font used by \.{TRIP.TEX}.
  There are no graphic characters associated with \.{trip} that could
  possibly be printed; indeed, \.{TRIP.PL} describes the properties of a font
  that is as weird as the ``document'' described by \.{TRIP.TEX}.
  
! Appendix D is \.{TRIPIN.LOG}, a correct transcript file \.{TRIP.LOG}
  that results if \.{INITEX} is applied to \.{TRIP.TEX}. (\.{INITEX} is
  the name of a version of \TeX\ that does certain initializations;
  this run of \.{INITEX} also creates a binary format file called \.{TRIP.FMT}.)
  
! Appendix E is a correct transcript file \.{TRIP.LOG} that results if
  \.{INITEX} or any other version of \TeX\ is applied to \.{TRIP.TEX}
  with format \.{TRIP.FMT}.
  
! Appendix F is \.{TRIP.TYP}, the symbolic version of a correct output
  file \.{TRIP.DVI} that was produced at the same time as the \.{TRIP.LOG}
  file of Appendix~E.
  
! Appendix G is \.{TRIPOS.TEX}, a short file written out and read in
  by \TeX\ when it processes \.{TRIP.TEX}.
  
! Appendix H is \.{TRIP.FOT}, an abbreviated version of Appendix E that
  appears on the user's terminal during the run that produces \.{TRIP.LOG}
  and \.{TRIP.DVI}.
  
--- 223,261 ----
  The remainder of this report consists of appendices, whose contents ought
  to be described briefly here:
  
! \appref A explains in detail how to carry out a test of \TeX, given
  a tape that contains copies of the other appendices.
  
! \appref B is \.{TRIP.TEX}, the fiendish test file that has already
  been mentioned. People who think that they understand \TeX\ are challenged
  to see if they know what \TeX\ is supposed to do with this file.
  People who know only a little about \TeX\ might still find it
  interesting to study Appendix~B, just to get some insights into the
  methodology advocated here.
  
! \appref C is \.{TRIP.PL}, the property-list description of a
  special font called \.{trip}. This is the only font used by \.{TRIP.TEX}.
  There are no graphic characters associated with \.{trip} that could
  possibly be printed; indeed, \.{TRIP.PL} describes the properties of a font
  that is as weird as the ``document'' described by \.{TRIP.TEX}.
  
! \appref D is \.{TRIPIN.LOG}, a correct transcript file \.{TRIP.LOG}
  that results if \.{INITEX} is applied to \.{TRIP.TEX}. (\.{INITEX} is
  the name of a version of \TeX\ that does certain initializations;
  this run of \.{INITEX} also creates a binary format file called \.{TRIP.FMT}.)
  
! \appref E is a correct transcript file \.{TRIP.LOG} that results if
  \.{INITEX} or any other version of \TeX\ is applied to \.{TRIP.TEX}
  with format \.{TRIP.FMT}.
  
! \appref F is \.{TRIP.TYP}, the symbolic version of a correct output
  file \.{TRIP.DVI} that was produced at the same time as the \.{TRIP.LOG}
  file of Appendix~E.
  
! \appref G is \.{TRIPOS.TEX}, a short file written out and read in
  by \TeX\ when it processes \.{TRIP.TEX}.
  
! \appref H is \.{TRIP.FOT}, an abbreviated version of \appref E that
  appears on the user's terminal during the run that produces \.{TRIP.LOG}
  and \.{TRIP.DVI}.
  
***************
*** 384,390 ****
  \section Appendix F: The \.{TRIP.TYP} file.
  Here is another major component of the test. It shows the output of \.{DVItype}
  applied to the file \.{TRIP.DVI} that was created at the same time
! Appendix E was produced.
  \runninghead{APPENDIX F: \.{TRIP.TYP} (CONTINUED)}
  
  {\let\tt=\eighttt\leftskip 1in\baselineskip 9pt plus .1pt minus .1pt
--- 437,443 ----
  \section Appendix F: The \.{TRIP.TYP} file.
  Here is another major component of the test. It shows the output of \.{DVItype}
  applied to the file \.{TRIP.DVI} that was created at the same time
! \appref E was produced.
  \runninghead{APPENDIX F: \.{TRIP.TYP} (CONTINUED)}
  
  {\let\tt=\eighttt\leftskip 1in\baselineskip 9pt plus .1pt minus .1pt
***************
*** 395,401 ****
  
  \section Appendix G: The \.{TRIPOS.TEX} file.
  This short file was written out once and read in twice, during the time
! Appendix E was being produced. There are only three lines, the first of
  which is blank.
  \runninghead{APPENDIX G: \.{TRIPOS.TEX} (CONTINUED)}
  
--- 448,454 ----
  
  \section Appendix G: The \.{TRIPOS.TEX} file.
  This short file was written out once and read in twice, during the time
! \appref E was being produced. There are only three lines, the first of
  which is blank.
  \runninghead{APPENDIX G: \.{TRIPOS.TEX} (CONTINUED)}
  
***************
*** 404,413 ****
  \vfill\eject
  
  \section Appendix H: The \.{TRIP.FOT} file.
! This shows what appeared on the terminal while Appendix E was being produced.
  \runninghead{APPENDIX H: \.{TRIP.FOT} (CONTINUED)}
  
  \vskip8pt
  \verbatim{trip.fot}
  
! \vfill\end
--- 457,468 ----
  \vfill\eject
  
  \section Appendix H: The \.{TRIP.FOT} file.
! This shows what appeared on the terminal while \appref E was being produced.
  \runninghead{APPENDIX H: \.{TRIP.FOT} (CONTINUED)}
  
  \vskip8pt
  \verbatim{trip.fot}
  
! \vfill
! \input knuth-outline.tex
! \end
